cinder/tests/unit/volume/drivers/lightos/test_lightos_storage.py

```diff
@@ -0,0 +1,154 @@
+import http.client as httpstatus
+from unittest import mock
+
+# from cinder import context
+from cinder import exception
+# from cinder.i18n import _
+# from cinder.objects import fields
+from cinder.tests.unit import test
+from cinder.volume.drivers.lightos import constants, lightos
+from cinder.volume import configuration as conf
+
+FAKE_VOLUME_1 = {'size': 16,
+                 'name': 'FAKE_VOLUME_1',
+                 'id': 1,
+                 'status': 'available'}
+
+
+class InitiatorConnectorMock(object):
+
+    @staticmethod
+    def factory(protocol, root_helper, driver=None,
+                use_multipath=False,
+                device_scan_attempts=constants.DEVICE_SCAN_ATTEMPTS_DEFAULT,
+                arch=None,
+                *args, **kwargs):
+        return None
+
+
+class LightOSStorageVolumeDriverTest(test.TestCase):
+
+    def setUp(self):
+        """Initialize LightOS Storage Driver."""
+        super(LightOSStorageVolumeDriverTest, self).setUp()
+
+        configuration = mock.Mock(conf.Configuration)
+        # def append_config_values(config, opts):
+        #     for op in opts:
+        #         if not hasattr(config, op.name):
+        #             config.__setattr__(op.name, op.default)
+        # configuration.append_config_values = partial(append_config_values, configuration)
+
+        configuration.lightos_api_address = "10.10.10.71,10.10.10.72,10.10.10.73"
+        configuration.lightos_api_port=443
+        configuration.lightos_jwt=None
+        configuration.lightos_snapshotname_prefix = 'openstack_'
+        configuration.lightos_intermediate_snapshot_name_prefix = 'for_clone_'
+        configuration.lightos_default_compression_enabled=False
+        configuration.lightos_default_num_replicas=3
+        configuration.num_volume_device_scan_tries = constants.DEVICE_SCAN_ATTEMPTS_DEFAULT
+        configuration.lightos_api_service_timeout = constants.LIGHTOS_API_SERVICE_TIMEOUT
+        configuration.driver_ssl_cert_verify = False
+        # for some reason this value is not initialized by the driver parent configs
+        configuration.volume_name_template = 'volume-%s'
+
+        configuration.initiator_connector = "cinder.tests.unit.volume.drivers.lightos.test_lightos_storage.InitiatorConnectorMock"
+        # configuration.lightos_client = "cinder.tests.unit.volume.drivers.lightos.test_lightos_storage.LightOSConnectionMock"
+
+        self.driver = lightos.LightOSVolumeDriver(configuration=configuration)
+
+    # Yogev
+    def test_lightos_client_should_fail_if_lightos_api_address_empty(self):
+        """Test that the lightos_client validates api server endpoints."""
+
+        self.driver.configuration.lightos_api_address = ""
+        self.assertRaises(exception.InvalidParameterValue, lightos.LightOSConnection, self.driver.configuration)
+
+    # Yogev
```

Comment:
there are some local comments as well as commented out code that should be removed